__[graphdb and networkx](https://github.com/apowers313/roc/blob/master/experiments/2024.08.09-07.20.42-EXP-graphdb-and-networkx/2024.08.09-07.20.42-EXP-graphdb-and-networkx.ipynb)__

_Experiment Template v0.3.0_

# System Information

## Basic Info

In [1]:
%cd /home/apowers/Projects/roc
!date

/home/apowers/Projects/roc
Fri Aug  9 07:21:05 PDT 2024


## GIT

In [2]:
!echo "git hash:" `git rev-parse HEAD`
!echo "git status:"
!git status -s | cat
!echo

git hash: 09e5466b29bdabcea303453b7046a515b2451a11
git status:
 M roc/attention.py
 M roc/graphdb.py
 M roc/jupyter/state.py
 M tests/attention_test.py
 M tests/graphdb_test.py
?? ARCHITECTURE.md
?? experiments/2024.08.09-07.20.42-EXP-graphdb-and-networkx/
?? roc/feature_extractors/contains.py
?? roc/feature_extractors/ratio.py
?? roc/feature_extractors/shape.py
?? roc/jupyter/save.py



## Python

In [3]:
!python --version
!pip --version
!echo "installed packages:"
!pip list

Python 3.11.9
pip 23.1.2 from /home/apowers/Projects/roc/.venv/lib/python3.11/site-packages/pip (python 3.11)
installed packages:
Package                       Version           Editable project location
----------------------------- ----------------- --------------------------
alabaster                     0.7.16
annotated-types               0.7.0
astroid                       3.2.4
asttokens                     2.4.1
attrs                         24.1.0
Babel                         2.15.0
bandit                        1.7.9
black                         24.8.0
cachetools                    5.4.0
caerus                        0.1.10
certifi                       2024.7.4
cfgv                          3.4.0
chardet                       5.2.0
charset-normalizer            3.3.2
click                         8.1.7
cloudpickle                   3.0.0
colorama                      0.4.6
colored                       2.2.4
comm                          0.2.2
contourpy                    

## System

In [4]:
!uname -a
!echo
!echo "system cpus:"
!lscpu
!echo
!echo "system memory:"
!lsmem
!free

Linux 77efc77f71c7 5.15.0-116-generic #126-Ubuntu SMP Mon Jul 1 10:14:24 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

system cpus:
Architecture:             x86_64
  CPU op-mode(s):         32-bit, 64-bit
  Address sizes:          46 bits physical, 48 bits virtual
  Byte Order:             Little Endian
CPU(s):                   32
  On-line CPU(s) list:    0-31
Vendor ID:                GenuineIntel
  Model name:             Intel(R) Xeon(R) Gold 6134 CPU @ 3.20GHz
    CPU family:           6
    Model:                85
    Thread(s) per core:   2
    Core(s) per socket:   8
    Socket(s):            2
    Stepping:             4
    CPU max MHz:          3700.0000
    CPU min MHz:          1200.0000
    BogoMIPS:             6400.00
    Flags:                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge m
                          ca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 s
                          s ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc 
               

# ROC Setup

## ROC Config & Init

In [5]:
import roc

!echo ".env file:"
!cat .env
!echo

roc.init()
# or:
# roc.init({"setting": val})

print(f"config:\n{roc.Config.get()}")  # noqa: T201

.env file:
log_modules="graphdb:INFO;event:INFO"
config:
db_host='127.0.0.1' db_port=7687 db_conn_encrypted=False db_username='' db_password='' db_lazy=False node_cache_size=65536 edge_cache_size=65536 log_enable=True log_level='INFO' log_modules='graphdb:INFO;event:INFO' default_action='pass' perception_components=['delta:perception'] enable_gym_dump_env=False max_dump_frames=10 nethack_spectrum='chars'


## Import Paths

In [6]:
import sys

sys.path.append("./tests")

# Experiment

## NetworkX Export

In [1]:
from roc.graphdb import GraphDB

G = GraphDB.to_networkx()

/home/apowers/Projects/roc/roc/config.py:52: ConfigInitWarning: Getting settings before config module was initialized. Please call init() first
  warnings.warn(


In [3]:
G.nodes(data=True)

NodeDataView({0: {'name': 'Waymar Royce'}, 6: {'name': 'Nights Watch'}, 453: {'order': 1}, 7: {'name': 'Gared'}, 454: {'order': 2}, 4: {'imdb_rating': 9.1, 'name': 'Winter Is Coming', 'number': 1}, 5: {'number': 1}, 1: {'name': 'Beyond the Wall'}, 23: {'imdb_rating': 8.8, 'name': 'The Kingsroad', 'number': 2}, 460: {'order': 8}, 9: {'name': 'Winterfell'}, 455: {'order': 3}, 8: {'name': 'Will'}, 10: {'name': 'Ned Stark'}, 11: {'name': 'House Stark'}, 462: {'order': 10}, 477: {'order': 25}, 507: {'order': 55}, 1071: {'order': 619}, 1075: {'order': 623}, 29: {'name': 'Lady'}, 42: {'name': 'Lannister soldier'}, 363: {'name': 'Oswell Whent'}, 365: {'name': 'Arthur Dayne'}, 22: {'name': 'Summer'}, 677: {'order': 225}, 736: {'order': 284}, 1102: {'order': 650}, 21: {'name': 'Catspaw assassin'}, 49: {'name': 'Wildling'}, 247: {'name': 'Deer'}, 223: {'imdb_rating': 9.9, 'name': 'The Rains of Castamere', 'number': 9}, 178: {'number': 3}, 221: {'name': 'The Gift'}, 177: {'imdb_rating': 8.8, 'name

In [4]:
import networkx as nx

nx.write_gml(G, "test1.gml")

In [5]:
nx.write_gexf(G, "test1.gexf")

## %save

In [1]:
import roc

roc.init()

In [6]:
%save

format gexf
timestamp True
filename graph


In [5]:
%save -f gml --no-timestamp foo

format gml
timestamp False
filename foo


In [3]:
%save

Saving 2677 nodes...


Nodes: 100%|███████████████████████████| 2677/2677 [00:00<00:00, 11594.90node/s]

Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.235154


## Performance

### Before Node.get_many()
Saving 2677 nodes...
100%|██████████| 2677/2677 [01:49<00:00, 24.34it/s]
Saved 2677 nodes. Elapsed time: 0:01:49.975708

### After Node.get_many()
Saving 2677 nodes...
Nodes: 100%|██████████| 2677/2677 [00:03<00:00, 756.41node/s] 
Saved 2677 nodes. Elapsed time: 0:00:03.546128

## Formats

"gexf", "gml", "dot", "graphml", "json", "leda", "pajek", "matrix-market", "adj-list", "multi-adj-list", "edge-list"

In [1]:
import roc

roc.init()

In [2]:
%save -f gexf

Saving 2677 nodes...


Nodes:   0%|                                         | 0/2677 [00:00<?, ?node/s]

Nodes: 100%|█████████████████████████████| 2677/2677 [00:03<00:00, 762.31node/s]


Writing graph to 'graph-2024.08.11-16.19.14'...
Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:03.938639


In [3]:
%save -f gml

Saving 2677 nodes...


Nodes: 100%|███████████████████████████| 2677/2677 [00:00<00:00, 14874.64node/s]

Writing graph to 'graph-2024.08.11-16.19.15'...


Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.352954


In [4]:
%save -f dot

Saving 2677 nodes...


Nodes: 100%|████████████████████████████| 2677/2677 [00:00<00:00, 9475.87node/s]


Writing graph to 'graph-2024.08.11-16.19.16'...
Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.838471


In [5]:
%save -f graphml

Saving 2677 nodes...


Nodes: 100%|███████████████████████████| 2677/2677 [00:00<00:00, 15400.54node/s]

Writing graph to 'graph-2024.08.11-16.19.16'...


Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.658591


In [6]:
%save -f json-node-link

Saving 2677 nodes...


Nodes: 100%|███████████████████████████| 2677/2677 [00:00<00:00, 15227.67node/s]

Writing graph to 'graph-2024.08.11-16.19.17'...


Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.349104


In [7]:
%save -f json-adj

Saving 2677 nodes...


Nodes: 100%|███████████████████████████| 2677/2677 [00:00<00:00, 15630.61node/s]

Writing graph to 'graph-2024.08.11-16.19.17'...


Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.327570


In [8]:
%save -f cytoscape

Saving 2677 nodes...


Nodes: 100%|███████████████████████████| 2677/2677 [00:00<00:00, 15616.91node/s]

Writing graph to 'graph-2024.08.11-16.19.18'...


Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.465477


In [9]:
%save -f pajek

Saving 2677 nodes...


Nodes: 100%|███████████████████████████| 2677/2677 [00:00<00:00, 15700.66node/s]

Writing graph to 'graph-2024.08.11-16.19.18'...



/home/apowers/Projects/roc/.venv/lib/python3.11/site-packages/networkx/readwrite/pajek.py:75: UserWarning: Node attribute order is not processed. Non-string attribute.
  warnings.warn(
/home/apowers/Projects/roc/.venv/lib/python3.11/site-packages/networkx/readwrite/pajek.py:75: UserWarning: Node attribute imdb_rating is not processed. Non-string attribute.
  warnings.warn(
/home/apowers/Projects/roc/.venv/lib/python3.11/site-packages/networkx/readwrite/pajek.py:75: UserWarning: Node attribute number is not processed. Non-string attribute.
  warnings.warn(
/home/apowers/Projects/roc/.venv/lib/python3.11/site-packages/networkx/readwrite/pajek.py:93: UserWarning: Edge attribute count is not processed. Non-string attribute.
  warnings.warn(


Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.265294


In [10]:
%save -f matrix-market

Saving 2677 nodes...


Nodes: 100%|███████████████████████████| 2677/2677 [00:00<00:00, 15293.06node/s]

Writing graph to 'graph-2024.08.11-16.19.19'...


Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.384356


In [11]:
%save -f adj-list

Saving 2677 nodes...


Nodes: 100%|████████████████████████████| 2677/2677 [00:00<00:00, 9566.32node/s]

Writing graph to 'graph-2024.08.11-16.19.19'...
Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.294720


In [12]:
%save -f multi-adj-list

Saving 2677 nodes...


Nodes: 100%|███████████████████████████| 2677/2677 [00:00<00:00, 15098.70node/s]

Writing graph to 'graph-2024.08.11-16.19.19'...


Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.207788


In [13]:
%save -f edge-list

Saving 2677 nodes...


Nodes: 100%|███████████████████████████| 2677/2677 [00:00<00:00, 14888.47node/s]

Writing graph to 'graph-2024.08.11-16.19.20'...


Saved 2677 nodes and 11967 edges. Elapsed time: 0:00:00.223435
